### DISCUSSION TITLE
Is beanie safe-thread ?

### DISCUSSION THREAD

**Discussion by alm0ra** [BeanieODM/beanie#345] [Question]
Hey , i was lookijg for a safe thread library for mongodb in python .
I recently work with mongo engine based on pymongo but if you want use mongo engine in celery ,you must open a new connection 

Is beanie based on pymongo ? 
Is beanie safe threaded?
Is it possible you it in django with out async?

**Comment by roman-right** [comment]
Hey,

It is based on motor, which is based on pymongo.
I have to double-check this. If motor is thread-safe than yup.
I have a plan to add a sync version based on pymongo only which could be possible to use in sync applications. 1-2 weeks I'd say to release. For now, there are some workarounds for this, but I'd not recommend them.

**Reply by alm0ra** [reply]
Hey , thank you . We are waiting.
